app/Domain/Integrations/Repositories/EloquentIntegrationRepository.php

```diff
@@ -47,31 +48,44 @@ public function update(Integration $integration): void
             'type' => $integration->type,
             'name' => $integration->name,
             'description' => $integration->description,
-            'website' => $integration->website() ? $integration->website()->value : null,
+            'website' => $integration->website()?->value,
             'subscription_id' => $integration->subscriptionId,
             'status' => $integration->status,
             'partner_status' => $integration->partnerStatus,
             'key_visibility' => $integration->getKeyVisibility(),
-            'reminder_email_sent' => $integration->getReminderEmailSent(),
         ]);
     }
 
     /** @return Collection<Integration> */
-    public function getDraftsByTypeAndBetweenMonthsOld(IntegrationType $type, int $startMonths, int $endMonths): Collection
+    public function getDraftsByTypeAndBetweenMonthsOld(IntegrationType $type, int $startMonths, int $endMonths, string $mailType): Collection
     {
         return IntegrationModel::query()
             ->distinct()
             ->where('status', 'draft')
             ->where('type', $type->value)
-            ->whereNull('reminder_email_sent')
             ->whereBetween('created_at', [Carbon::now()->subMonths($endMonths), Carbon::now()->subMonths($startMonths)])
             ->has('contacts')  // This ensures that only integrations with at least one contact are returned
+            ->whereNotExists(function (QueryBuilder $query) use ($mailType) {
```

Comment:
I think you could add a relation to the `IntegrationModel` this could make this query easier to build and maintain. There is also a feature called `scopes` in Laravel. I think it's worth to have a look, even if not applicable to this situation.

Reply:
I added a relation and added scopes instead of the whereNotExists() check.